### DISCUSSION TITLE
Reasoning behind not supporting listening on the link-local address

### DISCUSSION THREAD

**Discussion by gr4yg00s3** [strongswan/strongswan#1862] [Q&A] [answered]
Hello, I see the following about link-local from https://wiki.strongswan.org/projects/strongswan/wiki/IPv6StrokeExamples: 

`Please be aware that the strongSwan IKE daemon cannot listen on IPv6 link-local addresses (fe80:..). You must assign a site-local, unique-local, or global IPv6 address to the physical network interface first.` 

The questions I had are:

1. What is the reasoning behind not supporting link-local?
2. Why doesn't the new config documentation https://docs.strongswan.org/docs/5.9/config/IPv6.html mention anything about this?

**Comment by tobiasbrunner** [comment]
No idea if such a limitation actually exists or what would cause it but I can see the potential for issues as link-local addresses are used for IPv6-internal operations e.g. NDP or DHCPv6, so protecting traffic to a link-local address might prevent these from working.

**Reply by gr4yg00s3** [reply]
Given that note is for the obsoloted stroke, I did try it with the latest and I see the below, so I'm presuming that limitation still exists:

```
Aug 24 05:01:22 09[IKE] <remote-peer-2|4> initiating IKE_SA remote-peer-2[4] to fe80::aa46:9dff:fe1b:10dc
Aug 24 05:01:22 09[IKE] <remote-peer-2|4> IKE_SA remote-peer-2[4] state change: CREATED => CONNECTING
Aug 24 05:01:22 09[CFG] <remote-peer-2|4> configured proposals: IKE:3DES_CBC/HMAC_SHA1_96/PRF_HMAC_SHA1/MODP_1024
Aug 24 05:01:22 09[CFG] <remote-peer-2|4> sending supported signature hash algorithms: sha256 sha384 sha512 identity
Aug 24 05:01:22 09[NET] <remote-peer-2|4> sending packet: from fe80::aa46:9dff:fe1b:efc[500] to fe80::aa46:9dff:fe1b:10dc[500] (332 bytes)
Aug 24 05:01:22 04[NET] error writing to socket: Invalid argument
Aug 24 05:01:26 06[IKE] <remote-peer-2|4> retransmit 1 of request with message ID 0
Aug 24 05:01:26 06[NET] <remote-peer-2|4> sending packet: from fe80::aa46:9dff:fe1b:efc[500] to fe80::aa46:9dff:fe1b:10dc[500] (332 bytes)
Aug 24 05:01:26 04[NET] error writing to socket: Invalid argument
```

**Reply by tobiasbrunner** [reply, marked as answer]
Sending from a link-local address requires specifying the interface to send from. You'll have to enable `charon.plugins.socket-default.set_sourceif` so it gets set automatically.

**Reply by gr4yg00s3** [reply]
Ah okay yes that's correct, let me try that. Thanks for pointing that out
